### PR TITLE
Make HelixCustomCodeRunner multi-ZK aware

### LINKED ISSUE #1242: HelixCustomCodeRunner doesn't have a multi-zk constructor
**Describe the bug**

HelixCustomCodeRunner doesn't have a multi-zk constructor

**To Reproduce**
Steps to reproduce the behavior.

N/A

**Expected behavior**

We need to have a constructor for multi-zk connection config.

**Additional context**

### PR DESCRIPTION
**Pull request by narendly**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1242

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This class was an edge case that was neglected as part of ZooScalability API migration. This commit makes HelixCustomCodeRunner realm-aware by adding a constructor that accepts a RealmAwareZkConnectionConfig.

### Tests

- [ ] The following tests are written for this issue:

None. Tests exist for CustomCodeRunner.

- [x] The following is the result of the "mvn test" command on the appropriate module:

mvn test on helix-core:
```
[ERROR] Failures: 
[ERROR]   TestZKCallback.testInvocation:223
[ERROR]   TestEnableCompression.testEnableCompressionResource:117 expected:<true> but was:<false>
[INFO] 
[ERROR] Tests run: 1162, Failures: 2, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```
Run separately:
```
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 27.043 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/participant/HelixCustomCodeRunner.java [hunk 1]
```diff
@@ -130,14 +147,23 @@ public void start() throws Exception {
...
-    HelixZkClient zkClient = null;
+    RealmAwareZkClient zkClient = null;
     try {
       // manually add ideal state for participant leader using LeaderStandby
       // model
-      HelixZkClient.ZkClientConfig clientConfig = new HelixZkClient.ZkClientConfig();
-      clientConfig.setZkSerializer(new ZNRecordSerializer());
-      zkClient = SharedZkClientFactory.getInstance()
-          .buildZkClient(new HelixZkClient.ZkConnectionConfig(_zkAddr), clientConfig);
+      if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED) || _zkAddr == null) {
```
Comment (dasahcc) [inline helix-core/src/main/java/org/apache/helix/participant/HelixCustomCodeRunner.java:154]: if zkaddress is null and we are initialize it with FederatedZKClient, do we have problem? For example, we cannot get the routing key and cause any problems?

Comment (narendly) [reply helix-core/src/main/java/org/apache/helix/participant/HelixCustomCodeRunner.java:154]: The whole point of FederatedZkClient is not using a particular ZkAddress, but rather rely on RealmAwareConnectionConfig to establish a multi-zk connection, so this makes sense. The sharding key information is included in the connection config.

**Comment by narendly** [comment]
This PR is ready to be merged.
